### PR TITLE
Attachment transfer progress indicators

### PR DESCRIPTION
**Pull request by mcginty**
Reused the "pending push" flag to indicator their upload state, which was previously only set for downloads.

When libtextsecure is updated I'll update the build.gradle file and checksums for new dependencies, I tested with a build installed locally in the mean time.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 1]
```diff
@@ -279,6 +279,7 @@ private void setStatusIcons(MessageRecord messageRecord) {
     mmsDownloadButton.setVisibility(View.GONE);
     mmsDownloadingLabel.setVisibility(View.GONE);
     indicatorText.setVisibility(View.GONE);
+    mediaThumbnail.setShowProgress(!messageRecord.isFailed());
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationItem.java]: should this not go in setMediaAttributes?

File: src/org/thoughtcrime/securesms/database/PartDatabase.java [hunk 2]
```diff
@@ -640,5 +651,22 @@ public String toString() {
...
+      if (o == null || getClass() != o.getClass()) return false;
+
+      PartId partId = (PartId)o;
+
+      if (rowId != partId.rowId) return false;
+      return uniqueId == partId.uniqueId;
+
+    }
+
+    @Override public int hashCode() {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/database/PartDatabase.java]: You can do Util.hashCode(rowId, uniqueId)

**Comment by moxie0** [comment]
handle download failure case?

File: src/org/thoughtcrime/securesms/database/PartDatabase.java (hunk 2 above)
Comment (mcginty) [inline src/org/thoughtcrime/securesms/database/PartDatabase.java]: done

**Comment by mcginty** [comment]
failure case handled

**Comment by moxie0** [comment]
in 2.21.0

**Pull request closed without merging**